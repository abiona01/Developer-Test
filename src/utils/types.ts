export type ArticleType = {
  postId: number;
  id: number;
  name: string;
  email: string;
  body: string;
};

export type ArticleFormType = {
  name: string;
  email: string;
  body: string;
};
