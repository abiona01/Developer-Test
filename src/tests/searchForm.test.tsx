import { describe, expect, it } from "vitest";
import { render, screen } from "../utils/test-utils";
import SearchForm from "../components/search-form";

describe("SearchForm", () => {
  it("should render", () => {
    expect(<SearchForm />).toMatchSnapshot();
  });
});

// The SearchForm component should disable the search button when the input value is empty
describe("SearchForm", () => {
  it("should disable the search button when the input value is empty", () => {
    render(<SearchForm />);
    const button = screen.getByRole("button", { name: "Search" });
    expect(button).toBeDisabled();
  });
});

// The SearchForm component should enable the search button when the input value is not empty
// describe("SearchForm", () => {
//   it("should enable the search button when the input value is not empty", () => {
//     render(<SearchForm />);
//     const input = screen.getByPlaceholderText("Search by name");
//     const button = screen.getByRole("button", { name: "Search" });
//     userEvent.type(input, "test");
//     expect(button).toBeEnabled();
//   });
// });
