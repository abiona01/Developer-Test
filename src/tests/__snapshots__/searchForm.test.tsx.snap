// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchForm > should render 1`] = `<SearchForm />`;
